### PR TITLE
18272 Add unit test for AboutTheBusiness component

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#18272

*Description of changes:*
- Added unit tests for AboutTheBusiness component

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: package.json
Comment (severinbeauvais) [inline package.json:3]: Please run `npm i` and commit the package lock file also.

File: tests/unit/AboutTheBusiness.spec.ts
Comment (severinbeauvais) [inline tests/unit/AboutTheBusiness.spec.ts:31]: Why would the elements not exist?

File: tests/unit/AboutTheBusiness.spec.ts
Comment (severinbeauvais) [inline tests/unit/AboutTheBusiness.spec.ts:37]: You shouldn't need the store anywhere in the tests, because the component itself doesn't use it.

File: tests/unit/AboutTheBusiness.spec.ts
Comment (severinbeauvais) [inline tests/unit/AboutTheBusiness.spec.ts:4]: Whenever possible, in unit tests, I think it's better to deep-mount the component under test.

Pro: Vuetify sub-components render and function correctly (they are not stubbed).

Con: All other sub-components are rendered and this may complicate the testing. Remember, we want to test THIS component, not everything under it (which should have their own tests). If you need a mix of Vuetify but not other sub-components, you can stub them out like this:

image

File: tests/unit/AboutTheBusiness.spec.ts
Comment (severinbeauvais) [inline tests/unit/AboutTheBusiness.spec.ts:25]: I think you don't need the router. Try without it.

File: tests/unit/AboutTheBusiness.spec.ts
Comment (severinbeauvais) [inline tests/unit/AboutTheBusiness.spec.ts:47]: This is fine, but another possibility, to avoid incomplete objects, might be:
```
data: {
  ...EmptyAgmExtEval,
  isGoodStanding: businessStore.isGoodStanding,
  incorporationDate: businessStore.getFoundingDate
} as AgmExtEvalIF
```

File: tests/unit/AboutTheBusiness.spec.ts
Comment (severinbeauvais) [inline tests/unit/AboutTheBusiness.spec.ts:1]: I should merge [my PR](#568) first as I did a bit of work here (which you can reject since yours is more complete).

Comment (leodube-aot) [reply tests/unit/AboutTheBusiness.spec.ts:1]: If your PR is close to merge I don't mind if you merge first while I work through the comments on this one. I'll finish up my review of your PR right now.

File: tests/unit/AboutTheBusiness.spec.ts
Comment (leodube-aot) [reply tests/unit/AboutTheBusiness.spec.ts:47]: Nice, I definitely prefer this approach much more casting as unknown first.

File: tests/unit/AboutTheBusiness.spec.ts
Comment (leodube-aot) [reply tests/unit/AboutTheBusiness.spec.ts:31]: I think what happens is that since the `data` prop is required, passing `null` to it does not create the component. [EntityHeader.spec.ts](https://github.com/bcgov/business-filings-ui/blob/ba9afd9d7ce0d4a1339c4f15f099da2be8baf41d/tests/unit/EntityHeader.spec.ts#L24) has a similar test in it, but perhaps a better way of doing this is to check that the component elements do not exist (like `v-card`).

Comment (severinbeauvais) [reply tests/unit/AboutTheBusiness.spec.ts:31]: Ah, I see.

You probably don't need to test that the prop is not provided, as that will result in a compile error. I suggest testing with an empty data object (either `EmptyAgmExtEval` or `{}`).

Comment (severinbeauvais) [reply tests/unit/AboutTheBusiness.spec.ts]: Fixed and looks good. Thanks!

**Review by JazzarKarim — APPROVED** [review]
LGTM Leo! 👍

File: tests/unit/AboutTheBusiness.spec.ts
Comment (severinbeauvais) [inline tests/unit/AboutTheBusiness.spec.ts:49]: Can you check for an actual long string date here (eg, "January 1, 2024")?

**Comment by severinbeauvais** [comment]
Please get other approvals before I approve. Thanks!

File: tests/unit/AboutTheBusiness.spec.ts
Comment (severinbeauvais) [reply tests/unit/AboutTheBusiness.spec.ts]: Yes, you got the date right.

This is better because your are testing the output of `dateToPacificDate()`. Previously, any error would have cancelled out.

**Review by PaulGarewal — APPROVED** [review]
LGTM 👍 Reviewing the Unit Testing was really helpful  🙌

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**
